### ISSUE TITLE
AgPasture's root variables are doubled

### ISSUE THREAD

**Issue by rcichota** [APSIMInitiative/ApsimX#3283]
The root related variables (e.g. BelowGroundWt, RootN) give values that are twice the actual value. This can be seen in the simulation in \\Prototypes\AgPasture. The two species are set to be initialised with a totalWt of 600, but the outputs show it has about 1200 in the first day of simulation.
This problem was found and fixed while working on #3188, but as some other change in that commit cause problems with multi-zone calculations those changes were reversed, so this issue has to be (re-)fixed

**Comment by rcichota** [comment]
This is just to re-cap on this fix. The issue with doubling some output arose from the fact that there are two variable to hold the information about roots (```plantZoneRoots``` and ```rootZones```). However, ```rootZones``` is a list with its first element equal to ```plantZoneRoots```. Few properties were summing up all the elements in the list and ```plantZoneRoots```, causing the error, for example:
```
        public double BelowGroundWt
        {
            get
            {
                double dmTotal = plantZoneRoots.DMTotal;
                foreach (PastureBelowGroundOrgan root in rootZones)
                    dmTotal += root.DMTotal;
                return dmTotal;
            }
        }
```
As the first item in ```rootZones``` is ```plantZoneRoots```, I deleted the later and simple replaced with ``rootZones[0]```. I also renamed ``rootZones``` to ``roots```. Thus the code above becomes
```
        public double BelowGroundLiveWt
        {
            get
            {
                double dmTotal = 0.0;
                foreach (PastureBelowGroundOrgan root in roots)
                    dmTotal += root.DMLive;
                return dmTotal;
            }
        }
```
However it seems clear that the extra roots in zones are not really part of the plant, they simply signify the presence of roots. the roots don't grow or senesce in these additional zones.  So, the code above should not consider the additional zones, only the "home" zone, thus:
```
        public double BelowGroundLiveWt
        {
            get
            {
                double dmTotal = roots[0].DMLive;
                return dmTotal;
            }
        }
```

**Comment by rcichota** [comment]
The issue shown above affect few outputs, but internally it didn't cause any problem. However, a similar coding was used for computing availableSoilN, and this has repercussions, although they seem fairly small.  Probably the SoilArbitrator, ensuring that mass balance was not broken, have minimised the effect of this. I have now fixed this, as follow (at it was then at it is now):
```
        private double[] mySoilNO3Available
        {
            get
            {
                double[] available = plantZoneRoots.mySoilNO3Available;
                foreach (PastureBelowGroundOrgan root in rootZones)
                    available = MathUtilities.Add(available, root.mySoilNO3Available);
                return available;
            }
        }
--------
        private double[] mySoilNO3Available
        {
            get
            {
                double[] available = new double[nLayers];
                foreach (PastureBelowGroundOrgan root in roots)
                    for (int layer = 0; layer < nLayers; layer++)
                        available[layer] += root.mySoilNO3Available[layer];
                return available;
            }
        }
```

**Comment by hol430** [comment]
Fixed by #3295
